### ISSUE TITLE
Issue: Unnamed docker volumes

### ISSUE THREAD

**Issue by max06** [devcontainers/cli#478]
(This is probably in the wrong repository - I'm not sure where to put this, please move it to the right people)

Specifying a mountpoint without a source-attribute creates a named volume named `undefined`. According to the spec (and to the same mount used in a `devcontainer.json`), it should create an unnamed volume.

Example:
```jsonc
    "mounts": [
        {
            "target": "/home/vscode/.vscode-server",
            "type": "volume"
        }
    ]
```

ends up as:

```text
            {
                "Type": "volume",
                "Name": "undefined",
                "Source": "/var/lib/docker/volumes/undefined/_data",
                "Destination": "/home/vscode/.vscode-server",
                "Driver": "local",
                "Mode": "z",
                "RW": true,
                "Propagation": ""
            }
```

**Comment by samruddhikhandale** [comment]
Hi 👋 

Thanks for reporting. 

According to this [docker documentation](https://docs.docker.com/storage/#:~:text=Anonymous%20volumes%20are%20not%20given,within%20a%20given%20Docker%20host.),

>  Anonymous volumes are not given an explicit name when they are first mounted into a container, so Docker gives them a random name that is guaranteed to be unique within a given Docker host.

Looks like a bug in the CLI - if the `source` is omitted, then I expect a new volume with random unique name should be created for volume mounts. This should definitely not be named as `undefined`

**Comment by max06** [comment]
Hey 👋🏼 

if I'm not mistaken, your schema requires the `source`-parameter for mounts. The bug in the cli might be its result.

**Comment by max06** [comment]
Hey there, @samruddhikhandale 
not trying to pressure anyone, just interested if there's any progress on this topic?

**Comment by samruddhikhandale** [comment]
Thanks for the ping on this issue. 

@alexander-smolyakov can you help prioritize this issue? We need a bug fix for this issue for the dev containers CLI.

**Comment by max06** [comment]
Closing this issue as it was fixed in #537 some time ago. Passed right by me unnoticed. 
I just tested it, I can confirm the solution, and I'm very happy about it! Thank you very much! 🥳
